### ISSUE TITLE
UnhandledPromiseRejectionWarning: FirebaseError: Can not convert variables for Cloud Storage rules runtime

### ISSUE THREAD

**Issue by vbeffa** [firebase/firebase-tools#3285]
### [REQUIRED] Environment info

**firebase-tools:** 9.10.0
**firebase-js-sdk:** 8.4.1

**Platform:** macOS 10.15.7

### [REQUIRED] Test case

I am not sure of a reproducible test case, but according to https://firebase.google.com/docs/reference/admin/node/admin.auth.DecodedIdToken `email_verified` is an optional boolean property of the token. The code where this gets parsed, https://github.com/firebase/firebase-tools/blob/v9.10.0/src/emulator/storage/rules/runtime.ts#L311-L367, does not check for boolean types. I added some logging to that function and this is what my token looks like:

```
{
  uid: 'xxx',
  token: {
    email: 'test@test.com',
    email_verified: true,
    auth_time: 1618512326,
    user_id: 'xxx',
    firebase: { identities: [Object], sign_in_provider: 'password' },
    iat: 1618512326,
    exp: 1618515926,
    aud: 'foo',
    iss: 'https://securetoken.google.com/foo',
    sub: 'xxx'
  }
}
```

### [REQUIRED] Steps to reproduce

See above.

### [REQUIRED] Expected behavior

Correct parsing without an error thrown.

### [REQUIRED] Actual behavior

```
(node:16159) UnhandledPromiseRejectionWarning: FirebaseError: Can not convert variables for Cloud Storage rules runtime
    at toExpressionValue (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:263:11)
    at /Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:254:27
    at Array.forEach (<anonymous>)
    at toExpressionValue (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:253:26)
    at /Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:254:27
    at Array.forEach (<anonymous>)
    at toExpressionValue (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:253:26)
    at createAuthExpressionValue (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:275:16)
    at createRequestExpressionValue (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:293:28)
    at StorageRulesRuntime.verifyWithRuleset (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:185:114)
    at StorageRulesetInstance.verify (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/rules/runtime.js:31:29)
    at isPermitted (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/apis/firebase.js:21:54)
    at handleUpload (/Users/vbeffa/.cache/firebase/tools/lib/node_modules/firebase-tools/lib/emulator/storage/apis/firebase.js:396:29)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Comment by samtstern** [comment]
@abeisgoat can you take a look at this one?

**Comment by abeisgoat** [comment]
Just put up a fix for this in #3310
